go-controller/pkg/cni/helper_linux.go

```diff
@@ -817,3 +831,71 @@ func (pr *PodRequest) deletePorts(ifaces []string, podNamespace, podName string)
 		pr.deletePort(iface, podNamespace, podName)
 	}
 }
+
+func getLinkLocalIP(namespace, podName, netName string, getter PodAndNodeInfoGetter) (string, error) {
+	pod, err := getter.getPod(namespace, podName)
+	if err != nil {
+		return "", err
+	}
+
+	node, err := getter.getNode(pod.Spec.NodeName)
+	if err != nil {
+		return "", err
+	}
+
+	lrpAddress, err := util.ParseNodeGatewayRouterJoinNetwork(node, netName)
+	if err != nil {
+		return "", err
+	}
+
+	if lrpAddress.IPv6 == "" {
+		return "", nil
+	}
+
+	// TODO: Support ipv6 single stack (mac is generated with ipv6)
+	// LRP's mac is calcualted from the join subnet IPv4
+	lrpAddressIPv4, _, err := net.ParseCIDR(lrpAddress.IPv4)
+	if err != nil {
+		return "", err
+	}
+	raSrcMAC := util.IPAddrToHWAddr(lrpAddressIPv4)
+	return util.HWAddrToIPv6LLA(raSrcMAC).String(), nil
+}
+
+func setupIngressFilter(linkLocalIP string) error {
+	tableName := "ingress_filter"
+	chainName := "input"
+	deviceName := "ovn-udn1"
+
+	nft, err := knftables.New(knftables.NetDevFamily, tableName)
+	if err != nil {
+		return fmt.Errorf("failed to initialize table: %w", err)
+	}
+
+	tx := nft.NewTransaction()
+
+	tx.Add(&knftables.Table{})
+
+	tx.Add(&knftables.Chain{
+		Name:     chainName,
+		Type:     knftables.PtrTo(knftables.FilterType),
+		Hook:     knftables.PtrTo(knftables.IngressHook),
+		Priority: knftables.PtrTo(knftables.FilterPriority),
+		Device:   knftables.PtrTo(deviceName),
+	})
+
+	tx.Add(&knftables.Rule{
+		Chain: chainName,
+		Rule: knftables.Concat(
+			"icmpv6", "type", "nd-router-advert", "ip6", "saddr", "!=", linkLocalIP, "@th,48,16", "!=", "0", "drop",
```

Comment:
let's define `@th,48,16` as a properly named constant, so nobody's surprised by this option.